adapters/vrtcal/vrtcal.go

```diff
@@ -87,3 +92,26 @@ func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters
 	}
 	return bidder, nil
 }
+
+func getReturnTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
+	for _, imp := range imps {
+		if imp.ID == impID {
+			if imp.Banner != nil {
+				return openrtb_ext.BidTypeBanner, nil
+			}
+
+			if imp.Video != nil {
+				return openrtb_ext.BidTypeVideo, nil
+			}
+
+			return "", &errortypes.BadServerResponse{
+				Message: fmt.Sprintf("Unsupported return type for ID: \"%s\"", impID),
+			}
+		}
+	}
+
+	//Failsafe default in case impression ID is not found
+	return "", &errortypes.BadServerResponse{
```

Comment:
Can you please add test coverage for this case here?

Reply:
> Can you please add test coverage for this case here?

I've added supplemental test coverage the failsafe unmatched impression ID. Thanks.